Building against 20.08 and latest SDK

**Describe the bug**
Arnold delegate doesnt build against 20.08

**To Reproduce**
Steps to reproduce the behavior:
1.  build USD 20.08
2. Try to build Arnold delegate with sdk 6.0.3.1

**Expected behavior**
for arnold delegate to build

**Screenshots**
`Install file: "build/linux_x86_64/gcc_profile/usd-0.20.8_arnold-6.0.3.1/schemas/_usdArnold.so" as "/home/sue.sauer/Documents/Maya_arnold-usd/lib/python/UsdArnold/_usdArnold.so"
Install file: "plugInfo.json" as "/home/sue.sauer/Documents/Maya_arnold-usd/lib/usd/plugInfo.json"
Install file: "build/linux_x86_64/gcc_profile/usd-0.20.8_arnold-6.0.3.1/schemas/source/generatedSchema.usda" as "/home/sue.sauer/Documents/Maya_arnold-usd/lib/usd/usdArnold/resources/generatedSchema.usda"
Install file: "build/linux_x86_64/gcc_profile/usd-0.20.8_arnold-6.0.3.1/schemas/source/plugInfo.json" as "/home/sue.sauer/Documents/Maya_arnold-usd/lib/usd/usdArnold/resources/plugInfo.json"
Install file: "build/linux_x86_64/gcc_profile/usd-0.20.8_arnold-6.0.3.1/schemas/schema.usda" as "/home/sue.sauer/Documents/Maya_arnold-usd/lib/usd/usdArnold/resources/usdArnold/schema.usda"
Install file: "render_delegate/plugInfo.json" as "/home/sue.sauer/Documents/Maya_arnold-usd/plugin/hdArnold/resources/plugInfo.json"
g++ -o build/linux_x86_64/gcc_profile/usd-0.20.8_arnold-6.0.3.1/render_delegate/config.os -c -fPIC -fno-operator-names -std=c++14 -w -O3 -g -O0 -pg -fPIC -D_LINUX -DUSD_HAS_FULLSCREEN_SHADER -DHDARNOLD_EXPORTS -I/home/sue.sauer/Documents/arnold_run/arnold_sdk/include -I/home/sue.sauer/Documents/USD/include -I/usr/include -I/usr/include/python2.7 -Ibuild/linux_x86_64/gcc_profile/usd-0.20.8_arnold-6.0.3.1/render_delegate -Irender_delegate render_delegate/config.cpp
g++ -o build/linux_x86_64/gcc_profile/usd-0.20.8_arnold-6.0.3.1/render_delegate/constant_strings.os -c -fPIC -fno-operator-names -std=c++14 -w -O3 -g -O0 -pg -fPIC -D_LINUX -DUSD_HAS_FULLSCREEN_SHADER -DHDARNOLD_EXPORTS -I/home/sue.sauer/Documents/arnold_run/arnold_sdk/include -I/home/sue.sauer/Documents/USD/include -I/usr/include -I/usr/include/python2.7 -Ibuild/linux_x86_64/gcc_profile/usd-0.20.8_arnold-6.0.3.1/render_delegate -Irender_delegate render_delegate/constant_strings.cpp
g++ -o build/linux_x86_64/gcc_profile/usd-0.20.8_arnold-6.0.3.1/render_delegate/debug_codes.os -c -fPIC -fno-operator-names -std=c++14 -w -O3 -g -O0 -pg -fPIC -D_LINUX -DUSD_HAS_FULLSCREEN_SHADER -DHDARNOLD_EXPORTS -I/home/sue.sauer/Documents/arnold_run/arnold_sdk/include -I/home/sue.sauer/Documents/USD/include -I/usr/include -I/usr/include/python2.7 -Ibuild/linux_x86_64/gcc_profile/usd-0.20.8_arnold-6.0.3.1/render_delegate -Irender_delegate render_delegate/debug_codes.cpp
g++ -o build/linux_x86_64/gcc_profile/usd-0.20.8_arnold-6.0.3.1/render_delegate/instancer.os -c -fPIC -fno-operator-names -std=c++14 -w -O3 -g -O0 -pg -fPIC -D_LINUX -DUSD_HAS_FULLSCREEN_SHADER -DHDARNOLD_EXPORTS -I/home/sue.sauer/Documents/arnold_run/arnold_sdk/include -I/home/sue.sauer/Documents/USD/include -I/usr/include -I/usr/include/python2.7 -Ibuild/linux_x86_64/gcc_profile/usd-0.20.8_arnold-6.0.3.1/render_delegate -Irender_delegate render_delegate/instancer.cpp
g++ -o build/linux_x86_64/gcc_profile/usd-0.20.8_arnold-6.0.3.1/render_delegate/light.os -c -fPIC -fno-operator-names -std=c++14 -w -O3 -g -O0 -pg -fPIC -D_LINUX -DUSD_HAS_FULLSCREEN_SHADER -DHDARNOLD_EXPORTS -I/home/sue.sauer/Documents/arnold_run/arnold_sdk/include -I/home/sue.sauer/Documents/USD/include -I/usr/include -I/usr/include/python2.7 -Ibuild/linux_x86_64/gcc_profile/usd-0.20.8_arnold-6.0.3.1/render_delegate -Irender_delegate render_delegate/light.cpp
g++ -o build/linux_x86_64/gcc_profile/usd-0.20.8_arnold-6.0.3.1/render_delegate/material.os -c -fPIC -fno-operator-names -std=c++14 -w -O3 -g -O0 -pg -fPIC -D_LINUX -DUSD_HAS_FULLSCREEN_SHADER -DHDARNOLD_EXPORTS -I/home/sue.sauer/Documents/arnold_run/arnold_sdk/include -I/home/sue.sauer/Documents/USD/include -I/usr/include -I/usr/include/python2.7 -Ibuild/linux_x86_64/gcc_profile/usd-0.20.8_arnold-6.0.3.1/render_delegate -Irender_delegate render_delegate/material.cpp
g++ -o build/linux_x86_64/gcc_profile/usd-0.20.8_arnold-6.0.3.1/render_delegate/mesh.os -c -fPIC -fno-operator-names -std=c++14 -w -O3 -g -O0 -pg -fPIC -D_LINUX -DUSD_HAS_FULLSCREEN_SHADER -DHDARNOLD_EXPORTS -I/home/sue.sauer/Documents/arnold_run/arnold_sdk/include -I/home/sue.sauer/Documents/USD/include -I/usr/include -I/usr/include/python2.7 -Ibuild/linux_x86_64/gcc_profile/usd-0.20.8_arnold-6.0.3.1/render_delegate -Irender_delegate render_delegate/mesh.cpp
render_delegate/mesh.cpp: In member function ‘virtual void pxrInternal_v0_20__pxrReserved__::HdArnoldMesh::Sync(pxrInternal_v0_20__pxrReserved__::HdSceneDelegate*, pxrInternal_v0_20__pxrReserved__::HdRenderParam*, pxrInternal_v0_20__pxrReserved__::HdDirtyBits*, const pxrInternal_v0_20__pxrReserved__::TfToken&)’:
render_delegate/mesh.cpp:205:95: error: ‘struct pxrInternal_v0_20__pxrReserved__::PxOsdOpenSubdivTokens_StaticTokenType’ has no member named ‘catmark’; did you mean ‘catmullClark’?
         if (scheme == PxOsdOpenSubdivTokens->catmullClark || scheme == PxOsdOpenSubdivTokens->catmark) {
                                                                                               ^~~~~~~
scons: *** [build/linux_x86_64/gcc_profile/usd-0.20.8_arnold-6.0.3.1/render_delegate/mesh.os] Error 1
scons: building terminated because of errors.
`

**Used Software Versions**
 - Arnold:  6.0.3.1
 - USD:  20.08
 - Compiler: GCC 6.3.1
 - OS:  Centos 7.7
 - Any 3rd-party app: Maya,Houdini

**Additional context**
Built delegate fine against Houdini 18.0.499(USD 19.11), Trying to build delegate against USD 20.08 for Maya 2020 doesn't work

Thanks Sue

--- a/render_delegate/hdarnold.h
+++ b/render_delegate/hdarnold.h
@@ -35,6 +35,11 @@
 #define USD_HAS_UPDATED_RENDER_BUFFER
 #endif
 
+#if USED_USD_VERSION_GREATER_EQ(20, 5)
+/// Not blitting to a hardware buffer anymore, following the example of HdEmbree.
+#define USD_DO_NOT_BLIT
+#endif
+
 #if AI_VERSION_NUMBER >= 60003
 /// Using the core instancer procedural instead of inline instances.
 #define HDARNOLD_USE_INSTANCER

--- a/render_delegate/mesh.cpp
+++ b/render_delegate/mesh.cpp
@@ -43,6 +43,7 @@ PXR_NAMESPACE_OPEN_SCOPE
 TF_DEFINE_PRIVATE_TOKENS(_tokens,
     (st)
     (uv)
+    (catmark)
 );
 // clang-format on
 
@@ -202,7 +203,7 @@ void HdArnoldMesh::Sync(
         AiNodeSetArray(_shape.GetShape(), str::nsides, nsides);
         AiNodeSetArray(_shape.GetShape(), str::vidxs, vidxs);
         const auto scheme = topology.GetScheme();
-        if (scheme == PxOsdOpenSubdivTokens->catmullClark || scheme == PxOsdOpenSubdivTokens->catmark) {
+        if (scheme == PxOsdOpenSubdivTokens->catmullClark || scheme == _tokens->catmark) {
             AiNodeSetStr(_shape.GetShape(), str::subdiv_type, str::catclark);
         } else {
             AiNodeSetStr(_shape.GetShape(), str::subdiv_type, str::none);

--- a/render_delegate/render_pass.cpp
+++ b/render_delegate/render_pass.cpp
@@ -314,6 +314,11 @@ void HdArnoldRenderPass::_Execute(const HdRenderPassStateSharedPtr& renderPassSt
         }
     };
 
+#ifdef USD_DO_NOT_BLIT
+    TF_VERIFY(!aovBindings.empty(), "No AOV bindings to render into!");
+#endif
+
+#ifndef USD_DO_NOT_BLIT
     if (aovBindings.empty()) {
         // We are first checking if the right storage pointer is set on the driver.
         // If not, then we need to reset the aov setup and set the outputs definition on the driver.
@@ -341,6 +346,7 @@ void HdArnoldRenderPass::_Execute(const HdRenderPassStateSharedPtr& renderPassSt
             _fallbackPrimId.Allocate({_width, _height, 1}, HdFormatInt32, false);
         }
     } else {
+#endif
         // AOV bindings exists, so first we are checking if anything has changed.
         // If something has changed, then we rebuild the local storage class, and the outputs definition.
         // We expect Hydra to resize the render buffers.
@@ -509,7 +515,9 @@ void HdArnoldRenderPass::_Execute(const HdRenderPassStateSharedPtr& renderPassSt
                     : AiArrayConvert(static_cast<uint32_t>(aovShaders.size()), 1, AI_TYPE_NODE, aovShaders.data()));
             clearBuffers(_renderBuffers);
         }
+#ifndef USD_DO_NOT_BLIT
     }
+#endif
 
     const auto renderStatus = renderParam->Render();
     _isConverged = renderStatus != HdArnoldRenderParam::Status::Converging;
@@ -526,7 +534,9 @@ void HdArnoldRenderPass::_Execute(const HdRenderPassStateSharedPtr& renderPassSt
             }
         }
         // If the buffers are empty, we have to blit the data from the fallback buffers to OpenGL.
-    } else {
+    }
+#ifndef USD_DO_NOT_BLIT
+    else {
         // Clearing all AOVs if render was aborted.
         if (renderStatus == HdArnoldRenderParam::Status::Aborted) {
             clearBuffers(_fallbackBuffers);
@@ -569,6 +579,7 @@ void HdArnoldRenderPass::_Execute(const HdRenderPassStateSharedPtr& renderPassSt
         _compositor.Draw();
 #endif
     }
+#endif
 }
 
 bool HdArnoldRenderPass::_RenderBuffersChanged(const HdRenderPassAovBindingVector& aovBindings)

--- a/render_delegate/render_pass.h
+++ b/render_delegate/render_pass.h
@@ -102,10 +102,12 @@ private:
     AtNode* _closestFilter = nullptr;  ///< Pointer to the closest Arnold Filter.
     AtNode* _mainDriver = nullptr;     ///< Pointer to the Arnold Driver writing color, position and depth.
 
+#ifndef USD_DO_NOT_BLIT
 #ifdef USD_HAS_FULLSCREEN_SHADER
     HdxFullscreenShader _fullscreenShader; ///< Hydra utility to blit to OpenGL.
 #else
     HdxCompositor _compositor; ///< Hydra compositor to blit to OpenGL.
+#endif
 #endif
 
     GfMatrix4d _viewMtx; ///< View matrix of the camera.

